src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -246,13 +246,14 @@ private function getBundlesResourcesPaths(ContainerBuilder $container): array
 
     private function getResourcesToWatch(ContainerBuilder $container, array $resourcesPaths): array
     {
-        // Flex structure
+        $paths = array_unique(array_merge($resourcesPaths, $this->getBundlesResourcesPaths($container)));
+
+        // Flex structure (only if nothing specified)
         $projectDir = $container->getParameter('kernel.project_dir');
-        if (is_dir($dir = "$projectDir/config/api_platform")) {
-            $resourcesPaths[] = $dir;
+        if (is_dir($dir = "$projectDir/config/api_platform") && empty($paths)) {
+            $paths = array_unique(array_merge($paths, [$dir]));
```

Comment:
`$paths[] = $dir;` ? As `$paths` is already empty, calling a `array_unique` + `array_merge` seems waaay overkill.

Reply:
indeed !!

Reply:
FIxed